### PR TITLE
[PM-7306] Onboarding users to new UI

### PR DESCRIPTION
**Pull request by gbubemismith**
## 🎟️ Tracking
[PM-7306](https://bitwarden.atlassian.net/browse/PM-7306)

## 📔 Objective
Show a dialog introducing the new UI. This dialog will only appear for accounts created before the release date and will not be shown again after a user interacts with it.

## 📸 Screenshots

| Dark  | Light  | Nord   |
|---|---|---|
|image|image|image|

https://github.com/user-attachments/assets/a4d27f00-2ef3-49cc-a112-5e982be68a8c

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-7306]: https://bitwarden.atlassian.net/browse/PM-7306?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by danielleflinn — DISMISSED** [review]

File: apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      </div>
+      <span bitDialogTitle>
+        {{ "bitwardenNewLook" | i18n }}
+      </span>
+      <span bitDialogContent>
+        {{ "bitwardenNewLookDesc" | i18n }}
+      </span>
+
+      <ng-container bitDialogFooter>
+        <a bitButton href="#" buttonType="primary" bitDialogClose (click)="navigateToLink()">
```
Comment (shane-melton) [inline apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts:48]: ❓ Do we need the `href` attribute here?

Comment (shane-melton) [reply apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts:48]: ❓ Also, do we want the dialog to close when they click to navigate? I see we have it opening in a new window, should we keep the dialog open? I believe this only concerns the Popout as the popup will likely close automatically once the new window is opened. cc: @danielleflinn

**Review by shane-melton — CHANGES_REQUESTED** [review]

File: apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts (hunk 1 above)
Comment (gbubemismith) [reply apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts:48]: The href can be removed

Comment (danielleflinn) [reply apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts:48]: @shane-melton I think we are ok with closing the dialog when the user selects the "Learn more" link. The main thing we want to do here is inform the user that the new UI is expected, and by clicking the navigation link I expect the user will be informed enough that they don't need to / or expect to see the dialog again.

Comment (gbubemismith) [reply apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts:48]: I don't think you can get the onboarding dialog to show in the popout, cause clicking outside the dialog in the extension closes the dialog and sets the state

Comment (shane-melton) [reply apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts:48]: Sounds good to me!

File: apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    <path class="tw-fill-info-600" d="m10.792 34.793 3.156 13.814-.92.21L9.87 35.004l.921-.21ZM21.59 29.817l4.246 18.578-.508.12-.512.118L20.68 30.02l.91-.203Z"/>
+    <path class="tw-fill-text-headers" d="M64.287.59A.945.945 0 0 1 65.58.248c8.784 5.11 15.628 14.039 18.166 25.145 2.537 11.105.25 22.12-5.443 30.538a.945.945 0 0 1-1.565-1.059c5.398-7.98 7.587-18.46 5.166-29.058C79.48 15.215 72.958 6.726 64.629 1.882A.945.945 0 0 1 64.287.59Z"/>
+    <path class="tw-fill-info-600" d="M61.6 6.385a.472.472 0 0 1 .643-.18c7.245 4.067 12.949 11.44 15.055 20.66s.171 18.338-4.588 25.149a.472.472 0 0 1-.774-.542c4.603-6.587 6.49-15.431 4.441-24.397-2.048-8.965-7.59-16.113-14.596-20.047a.472.472 0 0 1-.18-.643Z"/>
+    <path class="tw-fill-text-headers" d="M57.804 11.193a.472.472 0 0 1 .604-.285c6.11 2.186 11.426 8.739 13.364 17.22 1.938 8.48-.006 16.693-4.56 21.315a.472.472 0 1 1-.672-.663c4.27-4.335 6.197-12.187 4.311-20.442-1.886-8.254-7.032-14.49-12.761-16.54a.472.472 0 0 1-.286-.605Z"/>
+  </g>
+  <defs>
+    <clipPath id="a">
+      <path fill="#fff" d="M0 0h86v74H0z"/>
```
[the remark this answers is not in the document]
Comment (gbubemismith) [reply apps/browser/src/vault/popup/components/vault-v2/vault-ui-onboarding/vault-ui-onboarding.component.ts:26]: I think the hex might be okay in this case cause It doesn't do much, but I am OK to change to tailwind classes to keep it consistent

**Review by shane-melton — APPROVED** [review]

**Review by danielleflinn — APPROVED** [review]

File: apps/browser/src/vault/popup/services/vault-ui-onboarding.service.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  VAULT_BROWSER_UI_ONBOARDING,
+  "dialogState",
+  {
+    deserializer: (obj) => obj,
+  },
+);
+
+@Injectable()
+export class VaultUiOnboardingService {
+  // TODO: Update this date to the release date of the new Browser UI
```
Comment (nick-livefront) [inline apps/browser/src/vault/popup/services/vault-ui-onboarding.service.ts:27]: ❓ Is there a ticket associated with this TODO? I just don't want it to get forgotten about

**Review by nick-livefront — APPROVED** [review]

File: apps/browser/src/vault/popup/services/vault-ui-onboarding.service.ts (hunk 3 above)
Comment (gbubemismith) [reply apps/browser/src/vault/popup/services/vault-ui-onboarding.service.ts:27]: Yeah we have a ticket tracking this

**Review by justindbaur — APPROVED** [review]

**Pull request merged**